src/dbnode/storage/namespace.go

```diff
@@ -601,17 +605,43 @@ func (n *dbNamespace) QueryIDs(
 	query index.Query,
 	opts index.QueryOptions,
 ) (index.QueryResult, error) {
+	var (
+		sp opentracing.Span
+		ok bool
+	)
+
+	ctx, sp, ok = ctx.StartTraceSpan(tracepoint.NSQueryIDs)
+	if ok {
+		sp.LogFields(
+			opentracinglog.String("query", query.String()),
+			opentracinglog.String("namespace", n.ID().String()),
+			opentracinglog.Int("limit", opts.Limit),
+			xopentracing.Time("start", opts.StartInclusive),
+			xopentracing.Time("end", opts.EndExclusive),
+		)
+
+		defer sp.Finish()
+	}
+
 	callStart := n.nowFn()
 	if n.reverseIndex == nil { // only happens if indexing is enabled.
 		n.metrics.queryIDs.ReportError(n.nowFn().Sub(callStart))
-		return index.QueryResult{}, errNamespaceIndexingDisabled
+		err := errNamespaceIndexingDisabled
+		if sp != nil {
+			sp.LogFields(opentracinglog.Error(err))
+		}
+		return index.QueryResult{}, err
 	}
 
 	if n.reverseIndex.BootstrapsDone() < 1 {
 		// Similar to reading shard data, return not bootstrapped
 		n.metrics.queryIDs.ReportError(n.nowFn().Sub(callStart))
+		err := errIndexNotBootstrappedToRead
+		if sp != nil {
+			sp.LogFields(opentracinglog.Error(err))
+		}
 		return index.QueryResult{},
-			xerrors.NewRetryableError(errIndexNotBootstrappedToRead)
+			xerrors.NewRetryableError(err)
 	}
 
 	res, err := n.reverseIndex.Query(ctx, query, opts)
```

Comment:
Hm, maybe should also `if err != nil && sp != nil { sp.LogFields(opentracinglog.Error(err)) }` here?